### ISSUE TITLE
Persistence bug. Serializing and deserializing 2 times in a row results in exception

### ISSUE THREAD

**Issue by IMRaziel** [genaray/Arch.Extended#49]
Serializing and deserializing to json 2 times in a row (without even doing any changes to the world) results in different JSON texts, and throws exception during second deserialization.

Arch version 1.2.7
Arch.Persistence version 1.0.3

Full code to reproduce:
```csharp
using Arch.Core;
using Arch.Persistence;

var _world = World.Create();
var a = _world.Create(
    new Component { exp = 1, name = "c_1" }
);
var archSerializer = new ArchJsonSerializer();
var worldJson = archSerializer.ToJson(_world);
_world = archSerializer.FromJson(worldJson);
worldJson = archSerializer.ToJson(_world);
_world = archSerializer.FromJson(worldJson);


public struct Component
{
    public long exp;
    public string name;
}
```
Error message:
```csharp
System.IndexOutOfRangeException
  HResult=0x80131508
  Message=Index was outside the bounds of the array.
  Source=Arch.Persistence
  StackTrace:
   at Arch.Persistence.JaggedArrayFormatter`1.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Arch.Persistence.WorldFormatter.Deserialize(JsonReader& reader, IJsonFormatterResolver formatterResolver)
   at Arch.Persistence.ArchJsonSerializer.FromJson(String jsonWorld)
   at Program.<Main>$(String[] args) in C:\[...]\Program.cs:line 12
```

**Comment by genaray** [comment]
Thanks will look at this once i have time! :)

**Comment by bexogames** [comment]
Successfully reproduced with Arch-1.2.7 Arch.Persistence-1.0.3
But it seems already fixed in Arch-1.2.8 Arch.Persistence-1.0.4
